frontend/src/__tests__/cypress/cypress/pages/acceleratorProfile.ts

```diff
@@ -0,0 +1,240 @@
+    this.find().find(`[data-label="Toleration Seconds"]`).should('have.text', toleration);
+    return this;
+  }
+}
+
+class AcceleratorProfile {
+  visit() {
+    cy.visitWithLogin('/acceleratorProfiles');
+    this.wait();
+  }
+
+  sortedByColumn(column: string, order: string) {
+    return this.findColumn(column).then(($cells) => {
+      const cellValues = Array.from($cells).map((cell) => cell.textContent?.trim());
+      const sortedCellValues =
+        order === 'asc'
+          ? [...cellValues].sort()
+          : [...cellValues].sort((a, b) =>
+              b === undefined || a === undefined ? -1 : b.localeCompare(a),
+            );
+      expect(cellValues).to.deep.equal(sortedCellValues);
+    });
+  }
+
+  private wait() {
+    this.findAppPage();
+    cy.testA11y();
+  }
+
+  findAppPage() {
+    return cy.findByTestId('app-page-title');
+  }
+
+  findAddButton() {
+    return cy.findByTestId('display-accelerator-modal-button');
+  }
+
+  findEmptyText() {
+    return cy.findByTestId('no-available-accelerator-profiles');
+  }
+
+  findCreateButton() {
+    return cy.findByTestId('create-accelerator-profile');
+  }
+
+  findTableHeaderButton(name: string) {
+    return this.findTable().find('thead').findByRole('button', { name });
+  }
+
+  findResetButton() {
+    return cy.findByRole('button', { name: 'Reset' });
+  }
+
+  findFilterMenuOption() {
+    return cy.findByTestId('filter-dropdown-select');
+  }
+
+  findSearchInput() {
+    return cy.findByRole('textbox', { name: 'Search input' });
+  }
+
+  private findTable() {
+    return cy.findByTestId('accelerator-profile-table');
+  }
+
+  findColumn(column: string) {
+    return this.findTable().find('tbody').find(`[data-label=${column}]`);
+  }
+
+  getRow(column: string, name: string) {
+    return new AcceleratorRow(() => this.findColumn(column).contains(name).parents('tr'));
+  }
+}
+
+class ManageAcceleratorProfile {
+  findAcceleratorNameInput() {
+    return cy.findByTestId('accelerator-name-input');
+  }
+
+  findIdentifierInput() {
+    return cy.findByTestId('accelerator-identifier-input');
+  }
+
+  findTolerationsModalEmptyState() {
+    return cy.findByTestId('tolerations-modal-empty-state');
+  }
+
+  findTolerationsButton() {
+    return cy.findByTestId('add-toleration-button');
+  }
+
+  findDescriptionInput() {
+    return cy.findByTestId('accelerator-description-input');
+  }
+
+  findSubmitButton() {
+    return cy.findByTestId('accelerator-profile-create-button');
+  }
+
+  findTable() {
+    return cy.findByTestId('toleration-table');
+  }
+
+  getRow(name: string) {
+    return new TolerationRow(() =>
+      this.findTable().find(`[data-label=Key]`).contains(name).parents('tr'),
+    );
+  }
+}
+
+class TolerationsModal extends Modal {
+  constructor(edit = false) {
+    super(edit ? 'Edit toleration' : 'Add toleration');
+  }
+
+  findTolerationKeyInput() {
+    return cy.findByTestId('toleration-key-input');
```

Comment:
Scope modal selectors to the modal context.
```suggestion
    return this.find().findByTestId('toleration-key-input');
```